File: server/src/main/java/cn/net/yunlou/bole/entity/Skill.java
package cn.net.yunlou.bole.entity;

import cn.net.yunlou.bole.common.BaseEntity;
import com.baomidou.mybatisplus.annotation.TableName;
import java.math.BigDecimal;
import java.time.LocalDate;
import lombok.*;

/**
 * FileName: Skill Description: Created By MR. WANG Created At 2025/11/19 13:31 Modified By Modified
 * At
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("t_skill")
public class Skill extends BaseEntity {

    private Long userId;

    private String name;

    private String level;

    private String category;

    private String description;

    private Integer proficiencyPercent;

    private BigDecimal experienceYears;

    private Boolean isCertified;

    private String certificateName;

    private LocalDate certificateDate;

    private String tags;

    private Boolean isPublic;

    private Integer sort;
}

File: app/src/types/skill.ts
export interface SkillForm {
  id?: number;
  name: string;
  level: string;
  category: string;
  description: string;
  proficiencyPercent: number;
  experienceYears: number;
  isCertified: boolean;
  certificateName: string | null;
  certificateDate: string | null;
  tags: string[]; // 改为数组类型
  isPublic: boolean;
}

export interface SkillQuery extends BodyQuery {

}


export interface SkillResult {
  id: number;
  createdAt: string;
  updatedAt: string;
  deleted: number;
  name: string;
  level: string;
  category: string;
  description: string;
  proficiencyPercent: number;
  experienceYears: number;
  isCertified: boolean;
  certificateName: string | null;
  certificateDate: string | null;
  tags: string[];
  isPublic: boolean;
  sort: number;
}